molter/help.py

```diff
@@ -0,0 +1,197 @@
+import functools
+import logging
+
+import dis_snek
+from dis_snek import Embed
+from dis_snek.ext.paginators import Paginator
+
+import molter
+
+__all__ = ("HelpCommand",)
+
+log = logging.getLogger(dis_snek.const.logger_name)
+
+
+class HelpCommand:
+    show_hidden: bool
+    """Should hidden commands be shown?"""
+    show_disabled: bool
+    """Should disabled commands be shown?"""
+    run_checks: bool
+    """Should commands be checked if they can be run by the help command user?"""
+    show_self: bool
+    """Should this command be shown in the help message?"""
+    show_params: bool
+    """Should parameters for commands be shown?"""
+    show_aliases: bool
+    """Should aliases for commands be shown?"""
+    show_prefix: bool
+    """Should the prefix be shown?"""
+
+    embed_title: str
+    """The title to use in the embed. {username} will be replaced by the client's username."""
+    not_found_message: str
+    """The message to send when a command is not found. {cmd_name} will be replaced by the requested command."""
+
+    _client: dis_snek.Snake
+
+    def __init__(
+        self,
+        client: dis_snek.Snake,
+        *,
+        show_hidden: bool = False,
+        run_checks: bool = False,
+        show_self: bool = False,
+        show_params: bool = False,
+        show_aliases: bool = False,
+        show_prefix: bool = False,
+        embed_title: str | None = None,
+        not_found_message: str | None = None,
+    ) -> None:
+        self._client = client
+        self.show_hidden = show_hidden
+        self.run_checks = run_checks
+        self.show_self = show_self
+        self.show_params = show_params
+        self.show_aliases = show_aliases
+        self.show_prefix = show_prefix
+        self.embed_title = embed_title or "{username} Help Command"
+        self.not_found_message = not_found_message or "Sorry! No command called `{cmd_name}` was found."
+        self.cmd = self._callback
+
+    def register(self) -> None:
+        """Register the help command in dis-snek."""
+        if not isinstance(self.cmd.callback, functools.partial):
+            # prevent wrap-nesting
+            self.cmd.callback = functools.partial(self.cmd.callback, self)
+
+        # replace existing help command if found
+        if "help" in self._client.commands:
+            log.warning("Replacing existing help command.")
+            del self._client.commands["help"]
+
+        self._client.add_message_command(self.cmd)  # type: ignore
+
+    async def send_help(self, ctx: dis_snek.MessageContext, cmd_name: str | None) -> None:
+        """
+        Send a help message to the given context.
+
+        args:
+            ctx: The context to use
+            cmd_name: An optional command name to send help for.
+        """
+        await self._callback.callback(ctx, cmd_name)  # type: ignore
+
+    @molter.msg_command(name="help")
+    async def _callback(self, ctx: dis_snek.MessageContext, cmd_name: str = None) -> None:
```

Comment:
`cmd_name` should be marked a keyword-only argument (`*, cmd_name`) so that users can enter in spaces and the like and still get commands.